script.js

```diff
@@ -1,100 +1,332 @@
+    costoTotalDOM.textContent = `$${costoTotalConGanancia.toFixed(2)}`;
+    costoGananciaDOM.textContent = `$${costoUnidad.toFixed(2)}`;
+
+    const modalFinal = new bootstrap.Modal(document.getElementById('calculosFinales'));
+    modalFinal.show();
+  }
+});
+
+// Vacía completamente la lista de materiales después de una confirmación del usuario
+document.getElementById("vaciar-materiales").addEventListener("click", function () {
+  if (calculadora.materiales.length === 0) {
+    return; 
   }
-    break;
   
-    case "4":
-      if (calculadora.materiales.length === 0 || calculadora.gastosFijos.length === 0) {
-          alert("Debe agregar al menos un material y un gasto fijo antes de calcular.");
-        } else if (gananciaPorcentual === null || cantidadProduccion === null) {
-          alert("Debe ingresar la ganancia y la cantidad a producir antes de calcular.");
-        } else {
-          const resumen = calculadora.generarResumen(gananciaPorcentual, cantidadProduccion);
-          alert(resumen);
-        }
-        break;
-
-         case "5":
-        salir = true;
-        alert("Gracias por usar CostoPro.");
-        break;
-
-      default:
-        alert("Opción inválida. Ingrese un número del 1 al 5.");
-        break;
-}
+  const filaConfirmacion = document.createElement("tr");
+  filaConfirmacion.innerHTML = `
+    <td colspan="4" class="text-center text-danger">
+      ¿Seguro que quieres vaciar todos los materiales?
+      <button class="btn btn-danger btn-sm btn-confirmar-vaciar">Confirmar</button>
+      <button class="btn btn-secondary btn-sm btn-cancelar-vaciar">Cancelar</button>
+    </td>
+  `;
+  tablaMateriales.innerHTML = "";
+  tablaMateriales.appendChild(filaConfirmacion);
+
+  filaConfirmacion.querySelector(".btn-confirmar-vaciar").addEventListener("click", () => {
+    calculadora.materiales = [];
+    guardarDatosEnStorage();
+    actualizarTablaMateriales();
+  });
+
+  filaConfirmacion.querySelector(".btn-cancelar-vaciar").addEventListener("click", () => {
+    actualizarTablaMateriales();
+  });
+});
+
+// Vacía completamente la lista de gastos fijos después de una confirmación del usuario
+document.getElementById("vaciar-gastos").addEventListener("click", function () {
+  if (calculadora.gastosFijos.length === 0) {
+    return;
   }
-}
-iniciarCalculadora(); 
\ No newline at end of file
+  
+  const filaConfirmacion = document.createElement("tr");
+  filaConfirmacion.innerHTML = `
+    <td colspan="3" class="text-center text-danger">
+      ¿Seguro que quieres vaciar todos los gastos fijos?
+      <button class="btn btn-danger btn-sm btn-confirmar-vaciar">Confirmar</button>
+      <button class="btn btn-secondary btn-sm btn-cancelar-vaciar">Cancelar</button>
+    </td>
+  `;
+  tablaGastosFijos.innerHTML = "";
+  tablaGastosFijos.appendChild(filaConfirmacion);
+
+  filaConfirmacion.querySelector(".btn-confirmar-vaciar").addEventListener("click", () => {
+    calculadora.gastosFijos = [];
```

Comment:
Delega la responsabilidad a calculadora:
```suggestion
calculadora.vaciarGastosFijos() 
```